pkg/operator/controllers/checker/internetchecker_test.go

```diff
@@ -68,38 +68,95 @@ var (
-	for _, test := range testCases {
-		t.Run(test.name, func(t *testing.T) {
-			client := &testClient{responses: test.responses}
-			err := r.checkWithRetry(client, urltocheck, 100*time.Millisecond)
-			if (err != nil) != test.wantError {
-				t.Errorf("InternetChecker.check() error = %v, wantErr %v", err, test.wantError)
+	ctx := context.Background()
+
+	var testCases = []struct {
+		name          string
+		responses     []*fakeResponse
+		wantErr       string
+		wantCondition operatorv1.ConditionStatus
+	}{
+		{
+			name:          "200 OK",
+			responses:     []*fakeResponse{okResp},
+			wantCondition: operatorv1.ConditionTrue,
+		},
+		{
+			name:          "bad request",
+			responses:     []*fakeResponse{badReq},
+			wantCondition: operatorv1.ConditionTrue,
+		},
+		{
+			name:          "eventual 200 OK",
+			responses:     []*fakeResponse{networkUnreach, timedoutReq, okResp},
+			wantCondition: operatorv1.ConditionTrue,
+		},
+		{
+			name:          "eventual bad request",
+			responses:     []*fakeResponse{timedoutReq, networkUnreach, badReq},
+			wantCondition: operatorv1.ConditionTrue,
+		},
+		{
+			name:          "timedout request",
+			responses:     []*fakeResponse{networkUnreach, timedoutReq, timedoutReq, timedoutReq, timedoutReq, timedoutReq},
+			wantErr:       "requeue",
+			wantCondition: operatorv1.ConditionFalse,
+		},
+	}
+
+	roleToConditionTypeMap := map[string]string{
+		operator.RoleMaster: arov1alpha1.InternetReachableFromMaster,
+		operator.RoleWorker: arov1alpha1.InternetReachableFromWorker,
+	}
+
+	for _, testRole := range []string{operator.RoleMaster, operator.RoleWorker} {
+		t.Run(testRole, func(t *testing.T) {
+			for _, test := range testCases {
+				t.Run(test.name, func(t *testing.T) {
+					arocli := arofake.NewSimpleClientset(
+						&arov1alpha1.Cluster{
+							ObjectMeta: metav1.ObjectMeta{
+								Name: arov1alpha1.SingletonClusterName,
+							},
+							Spec: arov1alpha1.ClusterSpec{
+								InternetChecker: arov1alpha1.InternetCheckerSpec{
+									URLs: []string{urltocheck},
+								},
+							},
+						},
+					)
+
+					r := &InternetChecker{
+						log:          utillog.GetLogger(),
+						role:         testRole,
+						checkTimeout: 100 * time.Millisecond,
+
+						arocli:     arocli,
+						httpClient: &testClient{responses: test.responses},
+					}
+					err := r.Check(ctx)
+					if err != nil && err.Error() != test.wantErr ||
+						err == nil && test.wantErr != "" {
+						t.Error(err)
+					}
+
+					instance, err := arocli.AroV1alpha1().Clusters().Get(ctx, arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+					if err != nil {
+						t.Error(err)
+					}
+
+					var condition operatorv1.OperatorCondition
+					for _, condition = range instance.Status.Conditions {
+						if condition.Type == roleToConditionTypeMap[testRole] {
+							break
+						}
+					}
+
+					if condition.Status != test.wantCondition {
+						t.Errorf(string(condition.Status))
+					}
```

Comment:
I would like to separate the code which makes actual http calls to the list of URLs from updating status of the operator.

But in this PR I'm just adding coverage for the exiting code with minimal changes (just to make it testable).

Reply:
This sounds like a good maintenance task when we get the opportunity next year